diesel/src/sqlite/connection/raw.rs

```diff
@@ -194,6 +242,76 @@ extern "C" fn run_custom_function<F>(
     }
 }
 
+#[allow(warnings)]
+extern "C" fn run_aggregator_step_function<'a, A>(
+    ctx: *mut ffi::sqlite3_context,
+    num_args: libc::c_int,
+    value_ptr: *mut *mut ffi::sqlite3_value,
+)
+    where A: Aggregator<&'a [*mut ffi::sqlite3_value]>,
+{
+    static NULL_DATA_ERR: &str = "An unknown error occurred. sqlite3_user_data returned a null pointer. This should never happen.";
+    static NULL_CONN_ERR: &str = "An unknown error occurred. sqlite3_context_db_handle returned a null pointer. This should never happen.";
+    static NULL_AG_CTX_ERR: &str = "An unknown error occurred. sqlite3_aggregate_context returned a null pointer. This should never happen.";
+
+    unsafe {
+        let data_ptr = ffi::sqlite3_user_data(ctx);
+        let data_ptr = data_ptr as *mut FnMut() -> A;
+        let f = match data_ptr.as_mut() {
+            Some(f) => f,
+            None => {
+                ffi::sqlite3_result_error(
+                    ctx,
+                    NULL_DATA_ERR.as_ptr() as *const _ as *const _,
+                    NULL_DATA_ERR.len() as _,
+                );
+                return;
+            }
+        };
+
+        // Grab the aggregator instance out of memory
+        let aggregate_context = ffi::sqlite3_aggregate_context(ctx, std::mem::size_of::<*mut A>() as i32);
+        let aggregate_context = aggregate_context as *mut *mut A;
+        let aggregator = match aggregate_context.as_mut() {
+            Some(a) => match a.as_mut() {
+                Some(a) => a,
+                None => {
+                    ffi::sqlite3_result_error(
+                        ctx,
+                        NULL_AG_CTX_ERR.as_ptr() as *const _ as *const _,
+                        NULL_AG_CTX_ERR.len() as _,
+                        );
+                    return;
+                }
+            },
+            None => {
+                ffi::sqlite3_result_error(
+                    ctx,
+                    NULL_AG_CTX_ERR.as_ptr() as *const _ as *const _,
+                    NULL_AG_CTX_ERR.len() as _,
+                );
+                return;
+            }
+        };
+
+        // If this is the first time calling sqlite3_aggregate_context, we will need to instantiate the aggregator
+        if aggregator.is_null() {
+            (*aggregator) = f();
+        }
+
+        let args = slice::from_raw_parts(value_ptr, num_args as _);
+        aggregator.step(args);
+    }
+}
+
+extern "C" fn run_aggregator_final_function<'a, A>(
+    ctx: *mut ffi::sqlite3_context,
+)
+    where A: Aggregator<&'a [*mut ffi::sqlite3_value]> + Send + 'static,
+{
+    // TODO @thekuom: fill this out
```

Comment:
We need to do the following here:

* Grab the aggregator context
* Get a reference to the aggregator, using the code from above
* Move out the existing aggregator and replace it with `OptionalAggregator::None`. That guarantees that we call a potential `Drop` impl while calling `Aggregator::finalize` without having a existing reference left behind somewhere.
* Create a `SerializedValue` using the same method as in `functions::register` line 39 - 51
* Write back the result to the database using the same method as in `run_custom_function` line 234 - 238